cmd/a2o-migrate/test/a2o-migrate.sh

```diff
@@ -0,0 +1,147 @@
+#!/bin/bash
+
+set -e
+
+ENGINE="balena"
+DOCKERDIR="/mnt/data/docker/"
+OVERLAYDIR="$DOCKERDIR/overlay2"
+AUFSDIR="$DOCKERDIR/aufs"
+
+# https://github.com/balena-os/balena-engine/blob/15e30e91a3426488c01823ca2a446b71e81ea263/daemon/graphdriver/overlay2/randomid.go#L19
+generateID () {
+	local _len=26 # https://github.com/balena-os/balena-engine/blob/15e30e91a3426488c01823ca2a446b71e81ea263/daemon/graphdriver/overlay2/overlay.go#L85
+	tr -dc 'A-Z0-9'  < /dev/urandom | fold -w "$_len" | head -n1
+}
+
+log () {
+	if [ "$1" = "ERROR" ]; then
+		echo "$2"
+		exit 1
+	fi
+	echo "$1"
+}
+
+clean_containers () {
+	log "Stopping systemd services ..."
+	systemctl stop update-resin-supervisor.timer
+	systemctl stop update-resin-supervisor
+	systemctl stop resin-supervisor
+	systemctl stop balena
+}
+
+if [ ! -d "$DOCKERDIR/aufs" ]; then
+	log ERROR "No aufs set in $DOCKERDIR ?."
+fi
+if [ -d "$DOCKERDIR/overlay2" ]; then
+	log ERROR "overlayfs2 already set in $DOCKERDIR ?."
+fi
+
+log "Using $DOCKERDIR as $ENGINE root directory."
+
+# TODO: handle containers as well
+clean_containers
+
+mkdir -p "$DOCKERDIR/overlay2/l"
+log "Migrating layers ..."
+for aufs_layerpath in "$DOCKERDIR"/aufs/diff/*; do
```

Comment:
Is this script an analog/alternative to the a2o tool?

Reply:
yes. this was the original bash implementation, I will remove it